app/scripts/services/widgets.js

```diff
@@ -28,7 +27,9 @@ angular.module('angularWidgetInternal')
         var original = service[method];
         service[method] = function () {
           if (arguments.length >= count && !$rootScope.$$phase) {
-            $rootScope.$evalAsync();
+            $timeout(function () {
+              $rootScope.$apply();
+            }, 0);
```

Comment:
actually we should just do `$rootScope.$applyAsync()` instead of `$rootScope.$evalAsync()`